provisioning/miniprov/hare_mp/main.py

```diff
@@ -254,17 +259,148 @@ def test(args):
+    logging.info('Check that all services are up in PCS.')
+
+    resp = execute(pcs_status)
+    logging.info('PCS status: %s', resp)
+
+    if 'cluster is not currently' in resp:
+        return
+
+    logging.info('Make Node ready for testing, by stopping the cluster.')
+    resp = execute(cluster_stop)
+    logging.info('Cluster Stopped: %s', resp)
+    if 'Cluster stop is in progress' not in resp:
+        logging.error('Cluster is in progress.')
+
+
+def test_hare_postreq(cdf_file: str, timeinfo: str, logfile: str):
+    """Test suite is for restoring the setting after test."""
+    pcs_status = ['pcs', 'status']
+    cluster_start = ['cortx', 'cluster', 'start']
+
+    logging.info('Start the Cluster')
+    resp = execute(cluster_start)
+    logging.info('cluster status: %s', resp)
+    if 'Cluster start operation performed' not in resp:
+        logging.error('Cluster not yet started.')
+
+    logging.info('PCS: Check all services are up.')
+    resp = execute(pcs_status)
+    logging.info('PCS status: %s', resp)
+    if 'stopped' not in resp:
+        logging.error('Some services are not up.')
+
+    logging.info('hctl: Check that all the services are started.')
+    cluster_sts = check_cluster_status(cdf_file)
+    if cluster_sts:
+        resp = execute(['journalctl', '--since', timeinfo, '>', logfile])
+        logging.info('created journal log: %s', logfile)
+        raise Exception('hctl status reports failure.')
+
+    logging.info('Successfully performed cleanup after testing')
+
+
+def _hctlstatus():
+    """know the hctl status."""
+    hctl_status = ['hctl', 'status', '-d']
+    hctl_shutdown = ['hctl', 'shutdown', '--skip-consul-stop']
+
+    process = subprocess.Popen(hctl_status,
+                               stdin=subprocess.PIPE,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               encoding='utf8')
+    logging.debug('Issuing CLI command: %s', hctl_status)
+    out, err = process.communicate()
+    exit_code = process.returncode
+    logging.debug('Finished. Exit code: %d', exit_code)
+    if exit_code:
+        if 'Cluster is not running' not in out:
+            resp = execute(hctl_shutdown)
+            logging.info('hctl shutdown: %s', resp)
+            if is_cluster_running():
+                raise Exception('After shutdown, hctl is running.')
+        else:
+            raise Exception(
+                f'Command hctl status exited with error code {exit_code}'
+                f'Command output: {err}')
+    else:
+        raise Exception('Fail to control cluster, exit the test.')
+
+
+def test_hare_bootstrap_shutdown(args):
+    """Test suite for single node hare init in loop."""
+    loop_count = int(args.dev[0])
+    hctl_shutdown = ['hctl', 'shutdown', '--skip-consul-stop']
+
+    test_hare_prereq()
+
+    if is_cluster_running():
+        _hctlstatus()
```

Comment:
As far as I understand, this function will just invoke `hctl status` and return the stdout as the return value. That value is ignored. Then what is the point to invoke this function here?